src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Cli.NuGetPackageDownloader;
+using Microsoft.DotNet.ToolPackage;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Workloads.Workload.Install
+{
+    internal class WorkloadManifestUpdater : IWorkloadManifestUpdater
+    {
+        private readonly IReporter _reporter;
+        private readonly IWorkloadManifestProvider _workloadManifestProvider;
+        private readonly INuGetPackageDownloader _nugetPackageDownloader;
+        private readonly string _userHome;
+
+        public WorkloadManifestUpdater(
+            IReporter reporter,
+            IWorkloadManifestProvider workloadManifestProvider,
+            INuGetPackageDownloader nugetPackageDownloader,
+            string userHome)
+        {
+            _reporter = reporter;
+            _workloadManifestProvider = workloadManifestProvider;
+            _userHome = userHome;
+            _nugetPackageDownloader = nugetPackageDownloader;
+        }
+
+        public async Task UpdateAdvertisingManifestsAsync(SdkFeatureBand featureBand)
+        {
+            var manifests = GetInstalledManifestIds();
```

Comment:
Why do the methods on this class take a `featureBand` parameter?  I think the workload manifest provider only provides access to the manifests for the current feature band.  So if you were to pass in a different `featureBand` to any of these methods I don't think things will work.

Is there any scenario where we'd be trying to do operations on a different `featureBand`?

Reply:
In VSMac we need that